warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:52:9
   |
LL | #![warn(x5400)] //~ WARN unknown lint: `x5400`
   |         ^^^^^
   |
note: the lint level is defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:39:28
   |
LL | #![warn(unused_attributes, unknown_lints)]
   |                            ^^^^^^^^^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:53:10
   |
LL | #![allow(x5300)] //~ WARN unknown lint: `x5300`
   |          ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:54:11
   |
LL | #![forbid(x5200)] //~ WARN unknown lint: `x5200`
   |           ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:55:9
   |
LL | #![deny(x5100)] //~ WARN unknown lint: `x5100`
   |         ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:110:8
   |
LL | #[warn(x5400)]
   |        ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:113:25
   |
LL |     mod inner { #![warn(x5400)] }
   |                         ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:116:12
   |
LL |     #[warn(x5400)] fn f() { }
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:119:12
   |
LL |     #[warn(x5400)] struct S;
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:122:12
   |
LL |     #[warn(x5400)] type T = S;
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:125:12
   |
LL |     #[warn(x5400)] impl S { }
   |            ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:129:9
   |
LL | #[allow(x5300)]
   |         ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:132:26
   |
LL |     mod inner { #![allow(x5300)] }
   |                          ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:135:13
   |
LL |     #[allow(x5300)] fn f() { }
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:138:13
   |
LL |     #[allow(x5300)] struct S;
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:141:13
   |
LL |     #[allow(x5300)] type T = S;
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:144:13
   |
LL |     #[allow(x5300)] impl S { }
   |             ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:148:10
   |
LL | #[forbid(x5200)]
   |          ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:151:27
   |
LL |     mod inner { #![forbid(x5200)] }
   |                           ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:154:14
   |
LL |     #[forbid(x5200)] fn f() { }
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:157:14
   |
LL |     #[forbid(x5200)] struct S;
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:160:14
   |
LL |     #[forbid(x5200)] type T = S;
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:163:14
   |
LL |     #[forbid(x5200)] impl S { }
   |              ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:167:8
   |
LL | #[deny(x5100)]
   |        ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:170:25
   |
LL |     mod inner { #![deny(x5100)] }
   |                         ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:173:12
   |
LL |     #[deny(x5100)] fn f() { }
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:176:12
   |
LL |     #[deny(x5100)] struct S;
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:179:12
   |
LL |     #[deny(x5100)] type T = S;
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:182:12
   |
LL |     #[deny(x5100)] impl S { }
   |            ^^^^^

warning: `#[macro_escape]` is a deprecated synonym for `#[macro_use]`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:411:17
   |
LL |     mod inner { #![macro_escape] }
   |                 ^^^^^^^^^^^^^^^^
   |
   = help: try an outer attribute: `#[macro_use]`

warning: `#[macro_escape]` is a deprecated synonym for `#[macro_use]`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:408:1
   |
LL | #[macro_escape]
   | ^^^^^^^^^^^^^^^

error: cannot find attribute `plugin_registrar` in this scope
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:45:4
   |
LL | #![plugin_registrar]
   |    ^^^^^^^^^^^^^^^^

error: cannot determine resolution for the attribute macro `test`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:225:3
   |
LL | #[test]
   |   ^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot determine resolution for the attribute macro `test`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:226:27
   |
LL | mod test { mod inner { #![test] }
   |                           ^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot determine resolution for the attribute macro `bench`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:240:3
   |
LL | #[bench]
   |   ^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot determine resolution for the attribute macro `bench`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:242:20
   |
LL |     mod inner { #![bench] }
   |                    ^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot determine resolution for the attribute macro `bench`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:244:7
   |
LL |     #[bench]
   |       ^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot determine resolution for the attribute macro `bench`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:247:7
   |
LL |     #[bench]
   |       ^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot determine resolution for the attribute macro `bench`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:250:7
   |
LL |     #[bench]
   |       ^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

warning: use of deprecated attribute `crate_id`: no longer used.
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:90:1
   |
LL | #![crate_id = "10"]
   | ^^^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[warn(deprecated)]` on by default

warning: use of deprecated attribute `no_start`: no longer used.
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:99:1
   |
LL | #![no_start]
   | ^^^^^^^^^^^^ help: remove this attribute

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:303:1
   |
LL |   #[no_mangle]
   |   ^^^^^^^^^^^^
...
LL | / mod no_mangle {
LL | |     //~^ NOTE not a function or static
LL | |     mod inner { #![no_mangle] }
LL | |     //~^ WARN attribute should be applied to a function or static [unused_attributes]
...  |
LL | |     //~| NOTE not a function or static
LL | | }
   | |_- not a function or static
   |
note: the lint level is defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:39:9
   |
LL | #![warn(unused_attributes, unknown_lints)]
   |         ^^^^^^^^^^^^^^^^^
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:470:1
   |
LL |   #[cold]
   |   ^^^^^^^
...
LL | / mod cold {
LL | |     //~^ NOTE not a function
LL | |
LL | |     mod inner { #![cold] }
...  |
LL | |     //~| NOTE not a function
LL | | }
   | |_- not a function
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a foreign function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:499:1
   |
LL |   #[link_name = "1900"]
   |   ^^^^^^^^^^^^^^^^^^^^^
...
LL | / mod link_name {
LL | |     //~^ NOTE not a foreign function or static
LL | |
LL | |     #[link_name = "1900"]
...  |
LL | |     //~| NOTE not a foreign function or static
LL | | }
   | |_- not a foreign function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:538:1
   |
LL |   #[link_section = "1800"]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^
...
LL | / mod link_section {
LL | |     //~^ NOTE not a function or static
LL | |
LL | |     mod inner { #![link_section="1800"] }
...  |
LL | |     //~| NOTE not a function or static
LL | | }
   | |_- not a function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:68:1
   |
LL | #![cold] //~ WARN attribute should be applied to a function
   | ^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a foreign function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:72:1
   |
LL | #![link_name = "1900"]
   | ^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:75:1
   |
LL | #![link_section = "1800"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:308:17
   |
LL |     mod inner { #![no_mangle] }
   |     ------------^^^^^^^^^^^^^-- not a function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:315:5
   |
LL |     #[no_mangle] struct S;
   |     ^^^^^^^^^^^^ --------- not a function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:320:5
   |
LL |     #[no_mangle] type T = S;
   |     ^^^^^^^^^^^^ ----------- not a function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:325:5
   |
LL |     #[no_mangle] impl S { }
   |     ^^^^^^^^^^^^ ---------- not a function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:476:17
   |
LL |     mod inner { #![cold] }
   |     ------------^^^^^^^^-- not a function
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:483:5
   |
LL |     #[cold] struct S;
   |     ^^^^^^^ --------- not a function
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:488:5
   |
LL |     #[cold] type T = S;
   |     ^^^^^^^ ----------- not a function
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:493:5
   |
LL |     #[cold] impl S { }
   |     ^^^^^^^ ---------- not a function
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a foreign function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:505:5
   |
LL |     #[link_name = "1900"]
   |     ^^^^^^^^^^^^^^^^^^^^^
...
LL |     extern "C" { }
   |     -------------- not a foreign function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
help: try `#[link(name = "1900")]` instead
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:505:5
   |
LL |     #[link_name = "1900"]
   |     ^^^^^^^^^^^^^^^^^^^^^

warning: attribute should be applied to a foreign function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:512:17
   |
LL |     mod inner { #![link_name="1900"] }
   |     ------------^^^^^^^^^^^^^^^^^^^^-- not a foreign function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a foreign function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:517:5
   |
LL |     #[link_name = "1900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^ ---------- not a foreign function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a foreign function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:522:5
   |
LL |     #[link_name = "1900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^ --------- not a foreign function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a foreign function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:527:5
   |
LL |     #[link_name = "1900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^ ----------- not a foreign function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a foreign function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:532:5
   |
LL |     #[link_name = "1900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^ ---------- not a foreign function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:544:17
   |
LL |     mod inner { #![link_section="1800"] }
   |     ------------^^^^^^^^^^^^^^^^^^^^^^^-- not a function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:551:5
   |
LL |     #[link_section = "1800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^ --------- not a function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:556:5
   |
LL |     #[link_section = "1800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^ ----------- not a function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

warning: attribute should be applied to a function or static
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:561:5
   |
LL |     #[link_section = "1800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^ ---------- not a function or static
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!

error: aborting due to 9 previous errors; 57 warnings emitted

For more information about this error, try `rustc --explain E0557`.
